be/src/vec/functions/function_string.h

```diff
@@ -386,6 +387,194 @@ class FunctionStrcmp : public IFunction {
+
+                auto idx = index_check_const(i, is_const_args[2]);
+                int size = current_offsets[idx] - current_offsets[idx - 1];
+                const char* tmp =
+                        reinterpret_cast<const char*>(&current_chars[current_offsets[idx - 1]]);
+                //split the string with '-'
+                std::string to_split_s(tmp, size);
+                std::vector<std::string> str_v = split(to_split_s.substr(0, 10), "-");
+
+                // check if the arg type is DATE|DATETIME
+                if (str_v.size() != 3) {
+                    return Status::Error<ErrorCode::INVALID_ARGUMENT>(
+                            "The range partition only support DATE|DATETIME");
+                }
+                int curr_len = 1;
+
+                res_data[res_offset[i - 1]] = 'p';
+                // day => 2022 12 12 000000
+                // month => 2022 12 01 000000
+                // year => 2022 01 01 000000
+                if (range_type[0] == 'd') {
+                    for (int j = 0; j < 3; j++) {
+                        memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[j].c_str(),
+                               str_v[j].size());
+                        curr_len += str_v[j].size();
+                    }
+                } else if (range_type[0] == 'm') {
+                    for (int j = 0; j < 2; j++) {
+                        memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[j].c_str(),
+                               str_v[j].size());
+                        curr_len += str_v[j].size();
+                    }
+                    memcpy(&res_data[res_offset[i - 1]] + curr_len, "01", 2);
+                    curr_len += 2;
+                } else if (range_type[0] == 'y') {
+                    memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[0].c_str(),
+                           str_v[0].size());
+                    curr_len += str_v[0].size();
+                    memcpy(&res_data[res_offset[i - 1]] + curr_len, "0101", 4);
+                    curr_len += 4;
+                } else {
+                    return Status::Error<ErrorCode::INVALID_ARGUMENT>(
+                            "The first argument is range, the second argument muse be "
+                            "day, month or year ");
+                }
+                memcpy(&res_data[res_offset[i - 1]] + curr_len, "000000", 6);
+                curr_len += 6;
+                res_offset[i] = res_offset[i - 1] + curr_len;
+            }
+        }
+        block.get_by_position(result).column = std::move(res);
+        return Status::OK();
+    }
+
+private:
+    std::u16string _string_to_u16string(const std::string& str) const {
+        std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> convert;
+        return convert.from_bytes(str);
+    }
+
+    std::string _string_to_unicode(const std::u16string& s) const {
+        std::string res_s;
+        if (s.length() > 0 && s[0] == '-') {
```

Comment:
what's this for? could you add some testcases to show that? I didnot find any testcase result with '_'